### PR TITLE
Add AGENTS.md file

### PR DESCRIPTION
**Pull request by GabyUnalaq**
# Description

This file is intended to provide background into the Python SDK to AI agents like Copilot. It includes general information about the purpose of the SDK and general insight into its development processes.

# Definition of Done

The PR shall be merged only if all items mentioned in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ank-sdk-python/blob/main/CONTRIBUTING.md#how-to-contribute) have been followed. In case an item is not applicable as described, please provide a short explanation in the description.

- [x] All steps in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ank-sdk-python/blob/main/CONTRIBUTING.md#how-to-contribute) have been handled

### CODE REVIEW

File: AGENTS.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+Control Interface to manage (start, stop, update) other workloads and read
+cluster state.
+
+The SDK is published on PyPI as `ankaios-sdk`. Current version and compatible
+Ankaios version are in `setup.cfg` under `[metadata]`.
+
+## 2) Non-negotiable rules
+
+- **Never break existing public API** — must keep semantic alignment with other Ankaios SDKs
+- **100% test coverage is enforced** — every new or changed code path must be covered
```
Comment (christoph-hamm) [inline AGENTS.md]: I strongly object this rule. It is stupid to setup such rules for human, as they will probably add shallow tests just to meet the goals (see Goodhart's Law). For LLM this dangerous. They will game the rules.

Write what we really want.
* Tests based on the expected behavior, not the code base
* A test for every corner cases
* A test for every error in every way this error can happen
* If multiple positive result type are possible (like for Option or other enums) at least one test for Some and None results or at least one for every enum variant which could be returned.

If you write the rules like this, while reviewing the result of the LLM, the code coverage might give you a hint, if some case has been missed.

File: AGENTS.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+```
+
+All commands accept extra arguments that are forwarded to the underlying tool
+(pytest / pylint / pycodestyle), e.g.:
+
+```sh
+python3 run_checks.py --utest tests/test_ankaios.py   # single file
+python3 run_checks.py --utest -k test_apply_workload  # by keyword
+```
+
```
Comment (christoph-hamm) [inline AGENTS.md]: Why is this information in the AGENTS.md file? This is something which is also useful for humans. I would put it in some file like `DEVELOPMENT.md`.

File: AGENTS.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+from ankaios_sdk import Ankaios
+
+with Ankaios() as ankaios:
+    state = ankaios.get_state(timeout=5)
+```
+
+Use `_ank_base` (from `ankaios_sdk._protos`) for proto message construction in
+tests. Never import `_pb2` files directly from outside the `_protos` package.
+User-facing code must not expose proto objects.
+
```
Comment (christoph-hamm) [inline AGENTS.md]: Same as above. Maybe `ARCHITECTURE.md`? Could also go to `DEVELOPMENT.md`.

File: AGENTS.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+## 7) Exceptions
+
+All exceptions derive from `AnkaiosException` (see `ankaios_sdk/exceptions.py`).
+
+## 8) Helper scripts (`tools/`, all support `--help`)
+
+- `fetch_protos.sh` — download proto files from a specific Ankaios branch
+- `update_version.sh` — bump SDK, Ankaios, and API versions across the project
+- `generate_docs.sh` — build Sphinx HTML docs into `docs/build/`
+- `install_ankaios.sh` — install Ankaios runtime (release or CI artifacts)
```
Comment (christoph-hamm) [inline AGENTS.md]: Also `DEVELOPMENT.md`.

File: AGENTS.md (hunk 1 above)
Comment (GabyUnalaq) [reply AGENTS.md]: Rewrote this.

File: AGENTS.md (hunk 2 above)
Comment (GabyUnalaq) [reply AGENTS.md]: This has been moved to `DEVELOPMENT.md`.

File: AGENTS.md (hunk 4 above)
Comment (GabyUnalaq) [reply AGENTS.md]: This has been moved to `DEVELOPMENT.md`.

File: AGENTS.md (hunk 3 above)
Comment (GabyUnalaq) [reply AGENTS.md]: This part is Ankaios specific, containing information an Ankaios developer should know already. The reason why is in here is to give an overview to the agent about the basic Control Interface functionality.

File: AGENTS.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
+# Agent Instructions
+
+## Purpose
+
+This project is the **Python SDK for Eclipse Ankaios** — a lightweight
+workload orchestrator for automotive embedded devices and HPCs. The SDK
+gives Python workloads running inside an Ankaios-managed container access
+to the Ankaios Control Interface to manage (start, stop, update) other
+workloads and read cluster state.
```
Comment (krucod3) [inline AGENTS.md]: Unify with Rust SDK:
```suggestion
Python SDK for Eclipse Ankaios — enables workloads inside an Ankaios cluster to start/stop/update
workloads, read state, stream logs, subscribe to events, and manage configs.
```

**Review by krucod3 — APPROVED** [review]
LGTM 👍
